library/std/src/net/endian.rs

```diff
@@ -0,0 +1,33 @@
+//! Define internal endianness primitives for networking.
+
+/// A Big Endian `[u8; 4]`.
+///
+/// Mostly used as inner type of `Ipv4Addr`.
+#[repr(transparent)]
+#[derive(Copy, Clone, PartialEq)]
+pub(super) struct BeU8x4([u8; 4]);
+
+impl BeU8x4 {
+    /// The argument `n` is interpreted in big endian order.
+    pub const fn from_be(n: u32) -> Self {
+        // Already in big endian order. See `to_be` for more explanation.
+        Self(n.to_ne_bytes())
```

Comment:
Now you are transmuting in one case but not in the other?

Reply:
`transmute` needs feature gate `#![feature(const_fn)]` but std didn't have one.
Somehow I cannot get it compile although I added the feature gate.
```
error[E0723]: can only call `transmute` from const items, not `const fn`
  --> library/std/src/net/endian.rs:29:18
   |
29 |         unsafe { transmute::<[u8; 4], u32>(self.0) }
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #57563 <https://github.com/rust-lang/rust/issues/57563> for more information
   = help: add `#![feature(const_fn)]` to the crate attributes to enable
```

Reply:
>    = help: add `#![feature(const_fn)]` to the crate attributes to enable

@oli-obk this seems like the wrong error? It should be `const_fn_transmute`.

Reply:
yea, we're pretty bad at these. After you add `const_fn` you get a feature error about `const_fn_transmute`